runtime/queries/metricsview_timeseries.go

```diff
@@ -72,73 +73,97 @@ func (q *MetricsViewTimeSeries) Resolve(ctx context.Context, rt *runtime.Runtime
 		return fmt.Errorf("metrics view '%s' does not have a time dimension", q.MetricsViewName)
 	}
 
-	// Build query
-	sql, args, err := q.buildMetricsTimeSeriesSQL(mv)
+	measures, err := toMeasures(mv.Measures, q.MeasureNames)
 	if err != nil {
-		return fmt.Errorf("error building query: %w", err)
+		return err
 	}
 
-	// Execute
-	meta, data, err := metricsQuery(ctx, olap, priority, sql, args)
+	tsq := &ColumnTimeseries{
+		TableName:           mv.Model,
+		TimestampColumnName: mv.TimeDimension,
+		TimeRange: &runtimev1.TimeSeriesTimeRange{
+			Start:    q.TimeStart,
+			End:      q.TimeEnd,
+			Interval: runtimev1.TimeGrain(runtimev1.TimeGrain_value["TIME_GRAIN_"+strings.ToUpper(q.TimeGranularity)]),
```

Comment:
Let's use a helper function for this with a switch statement to make it safer. Looks like there's already such a switch in `ColumnTimeGrain`.

Reply:
We need something more flexible than enum, we need to support arbitrary intervals like 15 mins, 5 mins, 2 weeks and so on.

FYI, I also noticed in UI code:
```
// The string values must adhere to DuckDB INTERVAL syntax, since, in some places, we interpolate an SQL queries with these values.
export enum TimeGrain {
  OneMinute = "minute",
  // FiveMinutes = "5 minute",
  // FifteenMinutes = "15 minute",
  OneHour = "hour",
  OneDay = "day",
  OneWeek = "week",
  OneMonth = "month",
  OneYear = "year",
}
```

Reply:
We should still use a helper function for mapping the enums.

But I get your point. Using non-single grains makes the SQL harder because you can't use date_trunc (afaik). Let's discuss with applications if there's a use case  that requires non-single grains (remember, grains and chart time ranges are not the same).

Reply:
Also, "interval" here is slightly a misnomer, as it's actually a "time grain"

Reply:
Yes, `date_trunc` cannot do it, but `extract` can help, ie https://stackoverflow.com/questions/68403015/is-there-a-way-to-group-by-intervals-of-15-min-in-duckdb